components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2822,6 +2823,25 @@ public static Certificate getX509CertOfOAuthApp(String clientId,
         }
     }
 
+    /**
+     * Return true if the token identifier is a parsable JWT.
+     *
+     * @param tokenIdentifier String JWT token identifier.
+     * @return true for a JWT token.
+     */
+    public static boolean isParsableJWT(String tokenIdentifier) {
+
+        if (StringUtils.isEmpty(tokenIdentifier)) {
+            return false;
+        }
+        try {
+            JWTParser.parse(tokenIdentifier);
+            return true;
+        } catch (ParseException e) {
+            return false;
```

Comment:
Lets add a debug log with the exception

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1380/commits/8b470323deb3712aede396c84307e8c6c5d305e9